### ISSUE TITLE
SQL query output as string and parameters.

### ISSUE THREAD

**Issue by gunnsth** [grindlemire/go-lucene#22]
For safety and to avoid SQL injection, it would seem beneficial to generate the SQL string as a string and all arguments as parameters.  Rather than just a single string.  This enables prepared statements where arguments are passed as placeholders with "?"

Example: For the following query
> color:red AND NOT (type:"honey crisp" OR type:"granny smith") AND age_in_months:[5 TO *]

It would generate a condition like:

```
    (
		("color" = ?) AND 
		(
			NOT(
				("type" = ?) OR 
				("type" = ?)
			)
		)
	) AND 
	("age_in_months" >= ?)
```
and a set of arguments
```
args = ["red", "honey crisp", "granny smith", 5]
```

That can be used as input to:
```
   db.Query("SELECT * FROM apples WHERE ...", args...)
```

As discussed here this provides additional security https://go.dev/doc/database/sql-injection that has been tested well.

It would probably also be good to validate the user-input fields (keys) and verify that they are intended to be accessible/filterable by the user.

Is it possible to achieve this with a custom driver possibly?

**Comment by grindlemire** [comment]
That is a good suggestion! More likely this could be a "mode" where you could render to parameterized input rather than a string. I'll chew on what that would look like and follow up here.

**Comment by grindlemire** [comment]
Ok I have a really rough implementation that works and has tests passing. It's pretty gross though so I am going to think of a way to clean it up. If you are curious you can see it at #23. I'll think about this some more and merge something in the next few days to add support here.

tl;dr is that during the render we build up a list of parameters and then replace literals with a ?. The tricky part that makes it gross is the type inference is lost for things like the range operator and general edge cases in the regular expression handling.

**Comment by grindlemire** [comment]
I just merged the PR so this is now exposed via the new `lucene.ToParameterizedPostgres` function. Thanks for filing the issue!
